chore(clerk-js): Remove sessions instead of destroying the client (#3941)
clerk/javascript: .changeset/funny-monkeys-brush.md

---
"@clerk/types": minor
"@clerk/clerk-js": minor
---

**Experimental:** Persist the Clerk client after signing out a user.
This allows for matching a user's device with a client. To try out this new feature, enable it in your `<ClerkProvider />` or `clerk.load()` call.

```js
// React

> <ClerkProvider  experimental={{ persistClient: true }} />

Reviewer: ```suggestion
<ClerkProvider experimental={{ persistClient: true }} />
```

---
"@clerk/types": minor
"@clerk/clerk-js": minor
---

**Experimental:** Persist the Clerk client after signing out a user.
This allows for matching a user's device with a client. To try out this new feature, enable it in your `<ClerkProvider />` or `clerk.load()` call.

```js
// React
<ClerkProvider  experimental={{ persistClient: true }} />

// Vanilla JS
await clerk.load({ experimental: { persistClient: true } })
```

> 
> Afterwards you can access `clerk.client.id` and `clerk.client.lastActiveSessionId`

Author: ```suggestion
```

---
"@clerk/types": minor
"@clerk/clerk-js": minor
---

> Experimental support: Persisting the client after signing a user out.
> `ClerkOptions` has been updated to accept `experimental: { persistClient: boolean }`

Reviewer 2: ````suggestion
**Experimental:** Persist the Clerk client after signing out a user.
This allows for matching a user's device with a client. To try out this new feature, enable it in your `<ClerkProvider />` or `clerk.load()` call.

```js
// React
<ClerkProvider  experimental={{ persistClient: true }} />

// Vanilla JS
await clerk.load({ experimental: { persistClient: true } })
```

Afterwards you can access `clerk.client.id` and `clerk.client.lastActiveSessionId`
````

Author: I'll remove "Afterwards you can access `clerk.client.id` and `clerk.client.lastActiveSessionId`" since this is not something that people should be interested in. This is more of a behind the scenes change, but we need the `experimental` prop in order to dogfood this for our apps but our clients to continue use the stable approach.
